The "1-my_list" module
======================

Using "MyList"
-------------------

>>> MyList = __import__('1-my_list').MyList

Normal cases:

>>> my_list = MyList()
>>> my_list.append(9)
>>> my_list.append(13)
>>> my_list.append(5)
>>> my_list.print_sorted()
[5, 9, 13]

>>> my_list = MyList()
>>> my_list.append(-3)
>>> my_list.append(3)
>>> my_list.remove(3)
>>> my_list.print_sorted()
[-3]

>>> my_list = MyList()
>>> my_list.print_sorted()
[]

>>> my_list = MyList()
>>> my_list.append("felipe")
>>> my_list.print_sorted()
['felipe']

>>> my_list = MyList()
>>> my_list.append(float('inf'))
>>> my_list.print_sorted()
[inf]

>>> my_list = MyList()
>>> my_list.append(None)
>>> my_list.print_sorted()
[None]

>>> my_list = MyList()
>>> my_list.append('a')
>>> my_list.print_sorted()
['a']

>>> my_list = MyList()
>>> my_list.append(3.545)
>>> my_list.print_sorted()
[3.545]

>>> my_list = MyList()
>>> my_list.append('A')
>>> my_list.append('B')
>>> my_list.append('C')
>>> my_list.print_sorted()
['A', 'B', 'C']

Error cases:

>>> my_list = MyList()
>>> my_list.append(5j)
>>> my_list.append(4)
>>> my_list.print_sorted()
Traceback (most recent call last):
TypeError: unorderable types: int() < complex()

>>> my_list = MyList()
>>> my_list.append(A)
Traceback (most recent call last):
NameError: name 'A' is not defined

>>> my_list = MyList(2, 3)
Traceback (most recent call last):
TypeError: list() takes at most 1 argument (2 given)

>>> my_list = MyList(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object is not iterable
